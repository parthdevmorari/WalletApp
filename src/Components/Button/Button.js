import React from 'react';
import { TouchableOpacity, View,Text } from 'react-native';
import styles from './ButtonStyle';



const Button = ({ title, onPress, style, textStyle,bottom, ...props }) => (
<View style={styles.submitBtnContainer}>
<TouchableOpacity onPress={onPress} {...props}style={[,bottom?styles.bottomBtn:styles.container,style]} >

		<Text style={[styles.btnText,textStyle]}>{title}</Text>

</TouchableOpacity>
</View>	
);

export default Button;
