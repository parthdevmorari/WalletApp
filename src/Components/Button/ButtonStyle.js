import { StyleSheet } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color';
import { theme } from "../../Constants/Theme";

const styles = StyleSheet.create({
	container: {
		backgroundColor:theme.action_Buttons_BG,
		borderRadius:3,
		paddingVertical:10,
		
	},
	submitBtnContainer: {
		width:'100%',
	},
	bottomBtn:{
		backgroundColor:color._Action_Buttons,
		borderRadius:3,
		paddingVertical:10,
		justifyContent:'flex-end',
		alignContent:'space-between',
		position:'absolute',
		bottom:wp(2),
		width:'100%',
	},
	btnText: {
		color:theme.action_Buttons_Text_Color,
		fontSize:wp(4),
		textAlign:'center',
		fontWeight:'700'
	}, 
	
	

	
});
export default styles;