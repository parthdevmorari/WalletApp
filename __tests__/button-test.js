import 'react-native';
import React from 'react';
import Button from '../src/Components/Button/Button'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('Button Component', () => {
  test('Button Component', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Button  toMatchSnapshot', () => {
    const wrapper = shallow(<Button />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});