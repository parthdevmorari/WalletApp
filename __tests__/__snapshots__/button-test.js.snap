// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button  toMatchSnapshot 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <TouchableOpacity
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#002A36",
          "borderRadius": 3,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
            "fontWeight": "700",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    />
  </TouchableOpacity>
</View>
`;
